### ISSUE TITLE
Checkout Remote doesn't checkout to remote branch

### ISSUE THREAD

**Issue by mkismy** [akonwi/git-plus#633]
When I "Checkout Remote" and select the remote branch to checkout, it doesn't checkout the branch but checkout the newest commit in the remote branch.

This happens after updating to 7.3.3

Steps to reproduce:

1. use "Checkout Remote"
2. select remote branch

After checking out, "Checkout" shows this:
![2017-03-02 12 09 58](https://cloud.githubusercontent.com/assets/8445575/23491369/32003b1e-ff41-11e6-828f-e3f98579be3a.png)

**Comment by nickhagen** [comment]
I am having this same issue, it appears from the command line the difference is calling:
`git checkout origin/<<branch>>`

vs 

`git checkout <<branch>>`

This is causing me some frustration, I can checkout the branch from the command line, but would love it if git-plus reverted that change.

**Comment by haoyangnz** [comment]
Is "Checkout Remote" supposed to track remote branch like "git checkout -b <branch> <remote>/<branch>"?

**Comment by nickhagen** [comment]
Previously it had functioned to checkout a remote branch and set the upstream ie:
`git checkout -b test origin/test`

Now it merely checks out a copy of the upstream branch in a detached state, ie:
`git checkout origin/test`

That is great for viewing the changes and running some tests, but if you have to edit anything you can't push those back up to origin.  I am alright with keeping that functionality if there is another way to create another command which does the same thing that it used to do would be very helpful.

**Comment by oppin** [comment]
Looks like the issue is in lib/models/git-checkout-branch.js

If you replace lines 13 and 14 with the following it should return to the original functionality.

```
const args = options.remote ? ['checkout', name, '-t'] : ['checkout', name]
git.cmd(args, {cwd: repo.getWorkingDirectory()})
```

**Comment by oppin** [comment]
I've submitted a pull request to incorporate this fix #637
